pkg/scheduler/queue/queue_test.go

```diff
@@ -844,9 +855,20 @@ func TestRequestQueue_tryDispatchRequestToQuerier_ShouldReEnqueueAfterFailedSend
 				require.NoError(t, queueBroker.enqueueRequestBack(&tr, tenantMaxQueriers))
 				require.False(t, tq.getNode(QueuePath{"tenant-1"}).IsEmpty())
 			} else if itq, ok := queueBroker.tree.(*MultiQueuingAlgorithmTreeQueue); ok {
-				require.Nil(t, itq.GetNode(QueuePath{"tenant-1"}))
+				var path QueuePath
+				if queueDim == nil {
+					queueDim = []string{unknownQueueDimension}
+				}
+				switch queueBroker.prioritizeQueryComponents {
+				case true:
```

Comment:
let's do an if-else on the boolean, I can't find examples of using a switch on a boolean anywhere else